### PR TITLE
Adds post functionality to web page 

### PR DESCRIPTION
**Pull request by nitinramadoss**
Adds form to html page and processes comments by storing them in servlet and returning them in JSON format to the client

### CODE REVIEW

File: portfolio/src/main/java/com/google/DataServlet.java [hunk 1]
```diff
@@ -24,30 +24,34 @@
...
 
    @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(phrases);
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+   @Override
```
Comment (jasperlee-google) [inline portfolio/src/main/java/com/google/DataServlet.java:43]: nit: think there is an extra space before `@Override`.

File: portfolio/src/main/java/com/google/DataServlet.java [hunk 2]
```diff
@@ -24,30 +24,34 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String message = request.getParameter("user-comment");
+    if(message == null || message.length() != 0){
+        comments.add(message);
```
Comment (jasperlee-google) [inline portfolio/src/main/java/com/google/DataServlet.java:47]: Do you want to add the `message` if its value is `null`?

File: portfolio/src/main/java/com/google/DataServlet.java [hunk 3]
```diff
@@ -24,30 +24,34 @@
...
+    }
+    response.sendRedirect("/index.html#comments-panel");
+  }
  
-  private String convertToJsonUsingGson(List<String> comments) {
+  private String convertToJsonUsingGson(List<String> content) {
```
Comment (jasperlee-google) [inline portfolio/src/main/java/com/google/DataServlet.java:52]: Think the fact that you are using `Gson` is an implementation detail and that `convertToJson` would be sufficient for the title.

For example, what if you need to refactor this later to use a different JSON lib?

File: portfolio/src/main/java/com/google/DataServlet.java [hunk 4]
```diff
@@ -24,30 +24,34 @@
...
-    String json = convertToJsonUsingGson(phrases);
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+   @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String message = request.getParameter("user-comment");
+    if(message == null || message.length() != 0){
```
Comment (lee12jin) [inline portfolio/src/main/java/com/google/DataServlet.java:46]: Spacing between if and the brace

if (message == null || message.length() != 0) {
}

Comment (nitinramadoss) [reply portfolio/src/main/java/com/google/DataServlet.java:46]: I'll add the spacing

File: portfolio/src/main/java/com/google/DataServlet.java (hunk 3 above)
Comment (nitinramadoss) [reply portfolio/src/main/java/com/google/DataServlet.java:52]: I hadn't thought of that, I'll change the function name

File: portfolio/src/main/java/com/google/DataServlet.java (hunk 2 above)
Comment (nitinramadoss) [reply portfolio/src/main/java/com/google/DataServlet.java:47]: Oh I made a mistake with the conditional, I'll fix that

File: portfolio/src/main/java/com/google/DataServlet.java (hunk 1 above)
Comment (nitinramadoss) [reply portfolio/src/main/java/com/google/DataServlet.java:43]: I placed only one space between each of the fields and methods of the DataServlet file

Comment (nitinramadoss) [reply portfolio/src/main/java/com/google/DataServlet.java:43]: Oh I see which space you were talking about, I'll fix that now

**Pull request merged**